app/assets/javascripts/media-views.js

```diff
@@ -0,0 +1,6487 @@
+				'main-video-playlist': 'mainVideoPlaylistToolbar',
+				'video-playlist-edit': 'videoPlaylistEditToolbar',
+				'video-playlist-add': 'videoPlaylistAddToolbar'
+			}
+		};
+
+		_.each( handlers, function( regionHandlers, region ) {
+			_.each( regionHandlers, function( callback, handler ) {
+				this.on( region + ':render:' + handler, this[ callback ], this );
+			}, this );
+		}, this );
+	},
+
+	activate: function() {
+		// Hide menu items for states tied to particular media types if there are no items
+		_.each( this.counts, function( type ) {
+			if ( type.count < 1 ) {
+				this.menuItemVisibility( type.state, 'hide' );
+			}
+		}, this );
+	},
+
+	mediaTypeCounts: function( model, attr ) {
+		if ( typeof this.counts[ attr ] !== 'undefined' && this.counts[ attr ].count < 1 ) {
+			this.counts[ attr ].count++;
+			this.menuItemVisibility( this.counts[ attr ].state, 'show' );
+		}
+	},
+
+	// Menus
+	/**
+	 * @param {techlogging.Backbone.View} view
+	 */
+	mainMenu: function( view ) {
+		view.set({
+			'library-separator': new techlogging.media.View({
+				className: 'separator',
+				priority: 100
+			})
+		});
+	},
+
+	menuItemVisibility: function( state, visibility ) {
+		var menu = this.menu.get();
+		if ( visibility === 'hide' ) {
+			menu.hide( state );
+		} else if ( visibility === 'show' ) {
+			menu.show( state );
+		}
+	},
+	/**
+	 * @param {techlogging.Backbone.View} view
+	 */
+	galleryMenu: function( view ) {
+		var lastState = this.lastState(),
+			previous = lastState && lastState.id,
+			frame = this;
+
+		view.set({
+			cancel: {
+				text:     'l10n.cancelGalleryTitle',
+				priority: 20,
+				click:    function() {
+					if ( previous ) {
+						frame.setState( previous );
+					} else {
+						frame.close();
+					}
+
+					// Keep focus inside media modal
+					// after canceling a gallery
+					this.controller.modal.focusManager.focus();
+				}
+			},
+			separateCancel: new techlogging.media.View({
+				className: 'separator',
+				priority: 40
+			})
+		});
+	},
+
+	playlistMenu: function( view ) {
+		var lastState = this.lastState(),
+			previous = lastState && lastState.id,
+			frame = this;
+
+		view.set({
+			cancel: {
+				text:     'l10n.cancelPlaylistTitle',
+				priority: 20,
+				click:    function() {
+					if ( previous ) {
+						frame.setState( previous );
+					} else {
+						frame.close();
+					}
+				}
+			},
+			separateCancel: new techlogging.media.View({
+				className: 'separator',
+				priority: 40
+			})
+		});
+	},
+
+	videoPlaylistMenu: function( view ) {
+		var lastState = this.lastState(),
+			previous = lastState && lastState.id,
+			frame = this;
+
+		view.set({
+			cancel: {
+				text:     'l10n.cancelVideoPlaylistTitle',
+				priority: 20,
+				click:    function() {
+					if ( previous ) {
+						frame.setState( previous );
+					} else {
+						frame.close();
+					}
+				}
+			},
+			separateCancel: new techlogging.media.View({
+				className: 'sep
```

Comment:
__[JSHint]__

Did you mean to return a conditional instead of an assignment?


<!-- Comment by [SideCI](https://sideci.com) -->